diarymanagement-code-r273/dbserver/branches/Dan-Test/src/TBA/Messages/LoginQuery.java

```diff
@@ -0,0 +1,148 @@
+package TBA.Messages;
+
+import TBA.Exceptions.MessagesException;
+
+/**
+ * This class is for the LOGIN query message.
+ *<p>
+ * @author Dan McGrath
+ *
+ * @version $Rev:: 27            $ $Date:: 2009-07-31 #$
+ */
+public class LoginQuery extends QueryMessage
+{
+   private String userName = null;
+   private String passwordHash = null;
+   private final int userNameLen = 10;
+   private final int passwordHashLen = 64;
+   private final int loginQueryLen = userNameLen + passwordHashLen;
+   private final String QueryType = "LOGIN";
+
+   public LoginQuery() { }
+
+   /**
+    * The Get() Method gets the LOGIN message with the Message and Query header.
+    *<p>
+    * @returns The actual message string you can send
+    *<p>
+    * @throws MessagesException
+    *<p>
+    * @see #Set(java.lang.String)
+    */
+   public String GetQuery() throws MessagesException
+   {
+      String message = null;
+
+      if(userName == null)
+      {
+         throw new MessagesException("Username not set");
+      }
+
+      message = String.format("%1$-" + Integer.toString(userNameLen) + "s", userName);
+
+      if(passwordHash.length() != passwordHashLen)
+      {
+         throw new MessagesException("Password Hash is not set or invalid");
+      }
+
+      message += passwordHash;
+      
+      super.SetQueryType(QueryType);
+      
+      return super.Get(message);
+   }
+
+   /**
+    * The SetQuery() Method strips out all the fields of a Login Query message. They
+    * are then available by the Get methods of this class.
+    *<p>
+    * @param message The message as recieved.
+    *<p>
+    * @throws MessagesException
+    *<p>
+    * @see #GetQuery(java.lang.String)
+    */
+   public void SetQuery(String message) throws MessagesException
+   {
+      message = super.Set(message);
+
+      if(message.length() != loginQueryLen)
+      {
+         throw new MessagesException("The Login Query is invalid");
+      }
+
+      userName = message.substring(0, userNameLen-1).trim();
+      passwordHash = message.substring(userNameLen, loginQueryLen);
+   }
+
+   /**
+    * This method returns the 'Username' field. You must have called either
+    * SetQuery() or SetUserName() first.
+    *<p>
+    * @returns The Username
+    *<p>
+    * @see #SetUserName(java.lang.String)
+    */
+   public String GetUserName()
+   {
+      return userName;
+   }
+
+   /**
+    * This method allows you to set the 'Username' field.
+    *<p>
+    * @param User The Username to set
+    *<p>
+    * @throws MessagesException
+    *<p>
+    * @see #GetUserName()
+    */
+   public void SetUserName(String User) throws MessagesException
+   {
+      if(userName == null)
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer

It seems unusual to set the value of a variable only when it is already set. You might want to switch the null check of == on line 102 to != check before setting the field value on line 109 or verify if the check is needed at all.